virtcontainers/device/drivers/utils.go

```diff
@@ -7,13 +7,97 @@
 package drivers
 
 import (
-	"github.com/sirupsen/logrus"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
 
 	"github.com/kata-containers/runtime/virtcontainers/device/api"
+	"github.com/kata-containers/runtime/virtcontainers/device/config"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	intMax uint = ^uint(0)
+
+	PCIDomain   = "0000"
+	PCIeKeyword = "PCIe"
 )
 
-const intMax uint = ^uint(0)
+type PCISysFsType string
+
+var (
+	PCISysFsDevices PCISysFsType = "devices" // /sys/bus/pci/devices
+	PCISysFsSlots   PCISysFsType = "slots"   // /sys/bus/pci/slots
+)
+
+type PCISysFsProperty string
+
+var (
+	PCISysFsDevicesClass     PCISysFsProperty = "class"         // /sys/bus/pci/devices/xxx/class
+	PCISysFsSlotsAddress     PCISysFsProperty = "address"       // /sys/bus/pci/slots/xxx/address
+	PCISysFsSlotsMaxBusSpeed PCISysFsProperty = "max_bus_speed" // /sys/bus/pci/slots/xxx/max_bus_speed
+)
 
 func deviceLogger() *logrus.Entry {
 	return api.DeviceLogger()
 }
+
+/*
+Identify PCIe device by /sys/bus/pci/slots/xx/max_bus_speed, sample content "8.0 GT/s PCIe"
+The /sys/bus/pci/slots/xx/address contains bdf, sample content "0000:04:00"
+bdf format: bus:slot.function
+*/
+func isVFIOPCIeDevice(bdf string) bool {
```

Comment:
As far as I can tell, this really does just detect a PCIe device right? I would suggest renaming this to isPCIeDevice instead.

Reply:
OK